mlflow/langchain/_langchain_autolog.py

```diff
@@ -31,6 +32,46 @@
 INFERENCE_FILE_NAME = "inference_inputs_outputs.json"
 
 
+# A *global* state that indicates whether MLflow should patch the inference method
+# for artifact auto-logging (model, signature input example). This disablement
+# is global across threads, as single model inference can trigger multiple threads,
+# for example, LangChain's batch()/abatch() API processes each request in a child thread.
+IS_PATCHING_DISABLED_FOR_ARTIFACTS = False
+
+# A *thread-local* state that indicates whether MLflow should patch the inference
+# method to inject tracing callbacks. This must be thread-local because MLflow can
+# run multiple model predictions across multiple threads (e.g. mlflow.evaluate)
+# and globally disabling tracing would cause issues for concurrent predictions.
+# In normal cases, we do not need this flag because MLflow checks if the call args
+# already contains the MLflow callback before injecting it, and LangChain propagates
+# callbacks from parent to child, preventing double patching. However, an exception
+# is LangGraph, where users might define their custom python function and invoke
+# some LangChain classes manually without propagating the callbacks.
+IS_PATCHING_DISABLED_FOR_TRACING = ContextVar("is_trace_disabled", default=False)
+
+
+@contextlib.contextmanager
+def disable_patching():
```

Comment:
```suggestion
def _disable_patching():
```
this should be used internally